addons/mrp/models/mrp_production.py

```diff
@@ -726,8 +726,8 @@ def _compute_move_finished_ids(self):
         for production in self:
             if production.state != 'draft':
                 updated_values = {}
-                if production.date_planned_finished:
-                    updated_values['date'] = production.date_planned_finished
+                if production.date_finished or production.date_planned_finished:
+                    updated_values['date'] = production.date_finished or production.date_planned_finished
```

Comment:
The date is set to `datetime.now` at the stock move validation (same for mrp.production). To me we should not update the finished move date if itis already in `done` state

Reply:
@Whenrow  do you mean that I should add and If for the state in the update of the `date` only or for the full update

Reply:
yes, we should not fill the `date` key in case of `state=='done'` stock moves